lecture_2/rgb_convert.c

```diff
@@ -0,0 +1,24 @@
+#define __STDC_FORMAT_MACROS
+#include <inttypes.h>
+#include <stdio.h>
+#include <stdint.h>
+
+
+uint16_t rgb_888_to_565(uint32_t rgb888);
+
+
+int main ()
+{
+    printf("RGB888 = 0x%X; RGB565 = 0x%X\n", 0x45a1, rgb_888_to_565(0x45a1));
+    
+    return 0;
+}
+
+uint16_t rgb_888_to_565(uint32_t rgb888)
```

Comment:
Your algorithm is very simplified and will distort colors.
read more about the conversions:
http://www.tune-it.ru/web/il/home/-/blogs/%D0%BF%D1%80%D0%B5%D0%BE%D0%B1%D1%80%D0%B0%D0%B7%D0%BE%D0%B2%D0%B0%D0%BD%D0%B8%D0%B5-rgb888-%3C-%3E-rgb565-%D0%B8-%D0%B7%D0%B0%D1%89%D0%B8%D1%82%D0%B0-%D0%B3%D1%80%D0%B8%D0%B1%D0%BE%D0%B2-%D0%BE%D1%82-%D0%B2%D1%8B%D1%86%D0%B2%D0%B5%D1%82%D0%B0%D0%BD%D0%B8%D1%8F;jsessionid=200eb623acd74674716a5f632999

Reply:
I saw this article. But at lection was mentioned that this exercise need to train me to work with bit shift operations.

Reply:
Ok.